pydantic/main.py

```diff
@@ -629,53 +621,68 @@ def _iter(
         self,
         to_dict: bool = False,
         by_alias: bool = False,
-        allowed_keys: Optional['SetStr'] = None,
         include: Union['AbstractSetIntStr', 'DictIntStrAny'] = None,
         exclude: Union['AbstractSetIntStr', 'DictIntStrAny'] = None,
         exclude_unset: bool = False,
         exclude_defaults: bool = False,
         exclude_none: bool = False,
     ) -> 'TupleGenerator':
 
+        allowed_keys = self._calculate_keys(include=include, exclude=exclude, exclude_unset=exclude_unset)
+        if allowed_keys is None and not any((to_dict, by_alias, exclude_unset, exclude_defaults, exclude_none)):
+            # huge boost for plain _iter()
+            yield from self.__dict__.items()
+            return
+
         value_exclude = ValueItems(self, exclude) if exclude else None
         value_include = ValueItems(self, include) if include else None
 
+        iteritems: Union[ItemsView[str, Any], 'TupleGenerator'] = self.__dict__.items()
+        # adding only needed checks/conversions to generator
+        if allowed_keys is not None:
+            iteritems = ((k, v) for k, v in iteritems if k in allowed_keys)
+        if exclude_none:
+            iteritems = ((k, v) for k, v in iteritems if v is not None)
         if exclude_defaults:
-            if allowed_keys is None:
-                allowed_keys = set(self.__fields__)
-            for k, v in self.__field_defaults__.items():
-                if self.__dict__[k] == v:
-                    allowed_keys.discard(k)
-
-        for k, v in self.__dict__.items():
-            if allowed_keys is None or k in allowed_keys:
-                value = self._get_value(
-                    v,
-                    to_dict=to_dict,
-                    by_alias=by_alias,
-                    include=value_include and value_include.for_element(k),
-                    exclude=value_exclude and value_exclude.for_element(k),
-                    exclude_unset=exclude_unset,
-                    exclude_defaults=exclude_defaults,
-                    exclude_none=exclude_none,
+            iteritems = ((k, v) for k, v in iteritems if self.__field_defaults__.get(k, _not_found) != v)
+        if by_alias:
+            fields = self.__fields__
+            iteritems = ((fields[k].alias if k in fields else k, v) for k, v in iteritems)
+        if to_dict or allowed_keys is not None:
```

Comment:
I guess even if we have one loop we could avoid calling `_get_value` in many cases?

Reply:
Yep. that's my point too 👍